sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -205,6 +236,25 @@ export abstract class V4PositionManager {
     const amount0Max = toHex(maximumAmounts.amount0)
     const amount1Max = toHex(maximumAmounts.amount1)
 
+    // Optionally, if the token is not native and has not been approved to Permit2, encode a permit call
+    if (options.token0Permit && !position.pool.token0.isNative) {
+      calldataList.push(ERC2612Permit.encodePermit(options.token0Permit))
+    }
+    if (options.token1Permit && !position.pool.token1.isNative) {
+      calldataList.push(ERC2612Permit.encodePermit(options.token1Permit))
+    }
+
+    // We use permit2 to approve tokens to the position manager
+    if (options.batchPermit) {
+      calldataList.push(
+        V4PositionManager.encodePermitBatch(
+          options.batchPermit.owner,
+          options.batchPermit.permitBatch,
+          options.batchPermit.signature
+        )
+      )
+    }
```

Comment:
I feel like we don't need to support permit2 permit for either token0 OR token1. We can just do a batchPermit of 1 element in that case, simplifying the options interface

Reply:
i wonder how much more gas it is. But yeah we can always do this as a follow-up? Maybe worth opening an issue or something just in case its way more gas

Reply:
Yep just made an issue #107